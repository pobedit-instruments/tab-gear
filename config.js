let date = new Date();

export default {
	email: 'podebit.ms@gmail.com',
	legal: 'Pobedit Instuments',
	year : date.getFullYear()
};
