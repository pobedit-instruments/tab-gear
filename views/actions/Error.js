import * as ActionTypes from '../constants/ActionTypes';

export const Error = {
	/**
	 * Show errors
	 *
	 * @returns {Object}
	 */
	showErrors () {
		return {
			type: ActionTypes.SHOW_ERRORS
		};
	}
};
