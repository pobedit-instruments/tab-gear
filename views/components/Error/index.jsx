import React, { Component, PropTypes } from 'react';

import Link from '../Link';
import Text from '../Text';

class Error extends Component {
	render () {
		let { email, legal } = this.props.config;

		return <Text>
					<p>
						{ chrome.i18n.getMessage('wrong') }
						<br />
						{ chrome.i18n.getMessage('contact') }

						<Link href={ `mailto:${email}` } mods={[ 'external' ]}>
							{ email }
						</Link>.
					</p>
				</Text>;
	}
}

Error.propTypes = {
	config: PropTypes.object.isRequired
};

export default Error;
