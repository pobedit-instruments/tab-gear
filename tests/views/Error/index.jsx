import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';

import config from '../../../config';
import Error from '../../../views/components/Error';

test('Error view', () => {
	let component = renderer.create(<Error config={ config } />);
	let view = component.toJSON();

	let actual = expect(view);

	actual.toMatchSnapshot();
});
