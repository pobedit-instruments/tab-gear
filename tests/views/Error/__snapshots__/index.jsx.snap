exports[`test Error view 1`] = `
<div
  className="tg-box">
   
  <p>
    Something went wrong!
    <br />
    Please contact us at 
    <a
      className="tg-link tg-link--external"
      href="mailto:podebit.ms@gmail.com"
      onClick={undefined}
      tabIndex={undefined}>
      podebit.ms@gmail.com
    </a>
    .
  </p>
   
</div>
`;
